### PR TITLE
Add color migration/autofill for legacy/empty ql charts

### PR DESCRIPTION
**Pull request by artemipanchuk**

### CODE REVIEW

File: src/server/modes/charts/plugins/ql/utils/migrate-helpers.ts [hunk 1]
```diff
@@ -6,96 +6,145 @@ import {isGroup} from './misc-helpers';
...
+    let xFields: Field[] = [];
     const yIndexes: number[] = [];
     const colorIndexes: number[] = [];
 
     let xDeclared = false;
     let yDeclared = false;
 
-    order.forEach((item: QlConfigResultEntryMetadataDataColumnOrGroup) => {
-        const itemIsGroup = isGroup(item);
+    if (order.length) {
```
Comment (kuzmadom) [inline src/server/modes/charts/plugins/ql/utils/migrate-helpers.ts:27]: I guess it's not necessary to check?

File: src/server/modes/charts/plugins/datalens/preparers/line-time/index.ts [hunk 2]
```diff
@@ -224,6 +225,14 @@ function prepareLineTime(options: PrepareFunctionArgs) {
             graph.color = colorData[i];
             graph.spanGaps = yPlaceholderSettings.nulls === 'connect';
         });
+
+        if (colorData.length > 1 && shared.extraSettings?.legendMode !== 'hide') {
```
Comment (kuzmadom) [inline src/server/modes/charts/plugins/datalens/preparers/line-time/index.ts:229]: The legend display should be based on the number of series with different names, not the presence of color

File: src/server/modes/charts/plugins/ql/utils/migrate-helpers.ts [hunk 3]
```diff
@@ -3,100 +3,142 @@ import type {QlConfigResultEntryMetadataDataColumnOrGroup} from '../../../../../
...
+const findIndexes = ({fields, forbiddenIndexes}: {fields: Field[]; forbiddenIndexes: number[]}) =>
+    fields
+        .map((field, fieldIndex) =>
+            (field.cast === DATASET_FIELD_TYPES.INTEGER ||
+                field.cast === DATASET_FIELD_TYPES.FLOAT) &&
+            !forbiddenIndexes.includes(fieldIndex)
+                ? fieldIndex
+                : null,
+        )
+        .filter((value) => value !== null) as number[];
```
Comment (kuzmadom) [inline src/server/modes/charts/plugins/ql/utils/migrate-helpers.ts:15]: to avoid going through the array twice, you can use reduce

**Review by kuzmadom — APPROVED** [review]

File: src/server/modes/charts/plugins/ql/utils/migrate-helpers.ts (hunk 3 above)
Comment (artemipanchuk) [reply src/server/modes/charts/plugins/ql/utils/migrate-helpers.ts]: Fixed

**Pull request merged**